### PR TITLE
neuron, nmodl: deploy latest NEURON with NMODL as default code generator for CoreNEURON

### PR DESCRIPTION
**Pull request by iomaganaris**
Ready for review after https://github.com/BlueBrain/nmodl/pull/1037 and https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/merge_requests/88 are merged

### CODE REVIEW

**Review by olupton — DISMISSED** [review]

**Review by pramodk — DISMISSED** [review]
LGTM

File: bluebrain/repo-bluebrain/packages/nmodl/package.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ class Nmodl(CMakePackage):
 
     version("develop", branch="master", submodules=True)
     version("llvm", branch="llvm", submodules=True)
+    version("0.6.0", tag="0.6")
     version("0.5.c", commit="c3b0736")
     version("0.5.b", commit="243453d")
     version("0.5.a", commit="94cba1c")
```
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/nmodl/package.py]: Can all these alpha versions go the way of the dodo?

That is the way.

File: bluebrain/repo-bluebrain/packages/sim-model/package.py [hunk 2]
```diff
@@ -146,7 +146,7 @@ def install(self, spec, prefix, install_src=True):
 
         bin/ <- special and special-core
         lib/ <- hoc, mod and lib*mech*.so
-        share/ <- neuron & coreneuron mod.c's (modc and modc_core)
+        share/ <- neuron & coreneuron mod.cpp's (modcpp and modcpp_core)
```
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/sim-model/package.py]:
```suggestion
        share/ <- neuron & coreneuron mod cpp files (modcpp and modcpp_core)
```
I'm left wondering what `mod.cpp` could possess.

File: bluebrain/repo-bluebrain/packages/nmodl/package.py (hunk 1 above)
Comment (iomaganaris) [reply bluebrain/repo-bluebrain/packages/nmodl/package.py]: This is the way.

**Review by matz-e — DISMISSED** [review]

**Comment by iomaganaris** [comment]
@matz-e I need your help with an issue I found out with this PR.
With these changes there are 2 different `neuron` packages deployed due to 2 different `nmodl` packages. This as a result makes neuron rebuild every time in the `blueconfigs` CI which is something we would like to avoid.
The issue is that `nmodl` depends on `nlohmann-json` which `spack` concertizes differently depending on whether we install `libsonata` with `neuron`. In the [libsonata recipe](https://github.com/BlueBrain/spack/blob/8d841647b912ff56a706ba3e8029db10862cc731/bluebrain/repo-bluebrain/packages/libsonata/package.py#L47) I saw you set a specific version limit while I can't find anything related to that in the `libsonata` repo. Maybe you have some more extra info on that? Do you think that removing this limitation in the version or adding the same in `nmodl` would fix the issue?
Here is the CI that rebuilds neuron: https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/jobs/685759

**Comment by iomaganaris** [comment]
With the latest changes blueconfigs CI use the deployed `neuron` version so everything good https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/jobs/689227
We're still validating and concertizing the reports in the blueconfigs with `nmodl` so this PR is not ready to merge

**Review by matz-e — DISMISSED** [review]

**Comment by iomaganaris** [comment]
blueconfigs CI is now passing with the updated reports and this can now be merged
https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/pipelines/121220

File: bluebrain/repo-patches/packages/neuron/package.py [hunk 3]
```diff
@@ -31,6 +31,7 @@ class Neuron(CMakePackage):
     patch("patch-v800-cmake-nvhpc.patch", when="@8.0.0%nvhpc^cmake@3.20:")
 
     version("develop", branch="master")
+    version("9.0.a6", commit="5665e19")
```
Comment (olupton) [inline bluebrain/repo-patches/packages/neuron/package.py]:
```suggestion
    version("9.0.a6", commit="67a672a")
```
neuronsimulator/nrn@67a672a2c802cb57e8a216d80af4b046f808bb02 would include https://github.com/neuronsimulator/nrn/pull/2350 and https://github.com/neuronsimulator/nrn/pull/2259. That would give us a fighting chance of being able to deploy `%oneapi` without a new `neuron` version.

Comment (olupton) [reply bluebrain/repo-patches/packages/neuron/package.py]: Just to make the explicit link, this ought to mean that https://github.com/BlueBrain/spack/pull/1891 would not need to add a new `neuron` version.

**Comment by iomaganaris** [comment]
blueconfigs CI after latest changes in https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/pipelines/121527 :heavy_check_mark: 
Ready to merge

**Review by olupton — APPROVED** [review]

**Pull request merged**